src/app.rs

```diff
@@ -286,14 +319,14 @@ where
         let _ = self.req_chan.blocking_send(AppReq::Render(delay));
         if let Some(response) = self.response_chan.blocking_recv() {
             let state =
-                if let Some(element) = self.root_pod.as_mut().and_then(|pod| pod.downcast_mut()) {
+                if let Some(element) = self.root_pod.as_mut() {
                     let mut state = response.state.unwrap();
                     let changes = response.view.rebuild(
                         &mut self.cx,
                         response.prev.as_ref().unwrap(),
                         self.id.as_mut().unwrap(),
                         &mut state,
-                        element,
+                        element.downcast_mut().unwrap(),
```

Comment:
Logically there are 3 cases:

1. `self.root_pod.as_mut()` is None
2. `self.root_pod.as_mut()` is Some and `element.downcast_mut()` is None
3. `self.root_pod.as_mut()` is Some and `element.downcast_mut()` is Some

Before, they are mapped to two arms: 3 maps to the true case, 1 & 2 maps to the false case.

Now, 3 is mapped to the true case, 1 is mapped to the false case, and 2 is mapped to a panic.

If we know out of band that 2 is impossible then there should be no change of behavior.

Reply:
That matches my thinking. I also did some minor digging and it seems `element.downcast_mut()` can only be `None` when the type assertion fails. However both the `build` and `rebuild` paths use `self.response_chan` which has a generic constraint on `View`. So my initial reading is that case#2 is impossible right now.

However I'm not super familiar with the Xilem type wrangling yet, hence my question. 🤔

Reply:
This was not intentional, but i think it is better this way. If the Widget type changes this is a bug which should be handled by the framework, instead of silently discarding everything and starting from scratch. That would make bugs much harder to detect.